src/sync/sync_permission.cpp

```diff
@@ -67,6 +67,22 @@ int64_t ns_since_unix_epoch(const system_clock::time_point& point)
 
 // MARK: - Permission
 
+Permission::Permission(Object& permission)
```

Comment:
I tried `const` but the implementation of `Object` won't allow it; there are a whole bunch of things on that class that would have to be audited and annotated :(.